src/zip2john.c

```diff
@@ -924,10 +924,18 @@ static void print_and_cleanup(zip_context *ctx)
 	filenames = xstrdup(ctx->best_files[0].file_name);
 	bname = jtr_basename(ctx->archive.fname);
 
-	printf("%s%s%s:$pkzip$%x*%x*", bname,
-			 ctx->num_candidates == 1 ? "/" : "",
-			 ctx->num_candidates == 1 ? ctx->best_files[0].file_name : "",
-			 ctx->num_candidates, ctx->archive.check_bytes);
+	if(ctx->num_candidates == 1) {
+		/* sanitize here, as we don't need the raw file_name any more when exiting this function */
+		replace(ctx->best_files[0].file_name, ':', ' ');
```

Comment:
Just a comment (I haven't tested this): I see we already replace with a space in line 957 in current code but IMHO we should replace with some non-space character such as '_' in both places.

Reply:
@magnumripper I think we should be consistent, and we already replace with spaces in other 2john tools. So let's do the same here for now. If we want to switch to another replacement character, we'll do so with a separate PR later. We also need to consider what will happen in `ldr_split_string`.